File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.cpp
#include <string>
#include <iostream>
using namespace std;

bool solution(string s)
{
    bool answer = true;
    int cntP =0, cntY = 0; 
    for(auto ss: s){
        if(ss=='p'||ss=='P'){
            cntP++;
        }
        if(ss=='y'||ss=='Y'){
            cntY++;
        }
    }
    if(cntP!=cntY){
        answer=false;
    }
    return answer;
}

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int nump = 0; 
        int numy = 0;
        for(char c : s.toCharArray()){
            if(c=='p'||c=='P') nump++;
            if(c=='y'||c=='Y') numy++;
        }
        
        if(nump!=numy){
            answer = false;
        }
        
        return answer;
    }
}